### PR TITLE
rlc deactivate completed

### PR DESCRIPTION
**Pull request by svandenb-dev**

### CODE REVIEW

File: pyaedt/edb_core/components.py [hunk 1]
```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
...
+    @pyaedt_function_handler()
+    def deactivate_rlc_component(self, component=None, create_circuit_port=False):
+        """Deactivate RLC component with possibility to replace them by circuit port.
+
+        Parameters
+        ----------
+        component : str
+            Component Refdes
+
+        create_circuit_port : bool
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:688]:
```suggestion
        create_circuit_port : bool, optional
```

File: pyaedt/edb_core/components.py [hunk 2]
```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
...
+        >>>     edb.core_components.deactivate_rlc_component(component=cmp, create_circuit_port=False)
+        >>> edb.save_edb()
+        >>> edb.close_edb()
+        """
+        if not component:
+            return False
+        if isinstance(component, str):
+            component = self.components[component]
+            if not component:
+                self._logger.error("component {} not found.".format(component))
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:711]: For the logging it is recommended to use the old `%s` for string format. This prevent the message to be evaluated if the log is not required.
```suggestion
                self._logger.error("component %s not found.", component)
```

File: pyaedt/edb_core/components.py [hunk 3]
```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
...
+                return False
+        if is_ironpython:
+            component_type = component.edbcomponent.GetComponentType()
+            if (
+                component_type == self._edb.Definition.ComponentType.Other
+                or component_type == self._edb.Definition.ComponentType.IC
+                or component_type == self._edb.Definition.ComponentType.IO
+            ):
+                self._logger.info("Component {} passed to deactivate is not an RLC.".format(component.refdes))
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:720]: For the logging it is recommended to use the old `%s` for string format. This prevent the message to be evaluated if the log is not required.
```suggestion
                self._logger.info("Component %s passed to deactivate is not an RLC.", component.refdes)
```

File: pyaedt/edb_core/components.py [hunk 4]
```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
...
+                return False
+        else:
+            if not component.edbcomponent.GetComponentType() in [1, 2, 3]:
+                self._logger.info("Component {} passed to deactivate is not an RLC.".format(component.refdes))
```
Comment (MaxJPRey) [inline pyaedt/edb_core/components.py:724]:
```suggestion
                self._logger.info("Component %s passed to deactivate is not an RLC.", component.refdes)
```

**Review by MaxJPRey — APPROVED** [review]

File: pyaedt/edb_core/components.py [hunk 5]
```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
...
+    def deactivate_rlc_component(self, component=None, create_circuit_port=False):
+        """Deactivate RLC component with possibility to replace them by circuit port.
+
+        Parameters
+        ----------
+        component : str
+            Component Refdes
+
+        create_circuit_port : bool
+            True, Component is converted into circuit port, False is just deactivated
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:689]:
```suggestion
            Whether to replace the deactivated RLC component to a circuit port. The default is ``False``.
```

File: pyaedt/edb_core/components.py [hunk 6]
```diff
@@ -676,6 +676,60 @@ def _get_closest_pin_from(self, pin, ref_pinlist):
...
+
+        Returns
+        -------
+        bool
+            True when succeeded, False when failed.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:694]:
```suggestion
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/edb_core/components.py [hunk 7]
```diff
@@ -686,7 +686,8 @@ def deactivate_rlc_component(self, component=None, create_circuit_port=False):
             Reference designator of the RLC component.
 
         create_circuit_port : bool
-            True, Component is converted into circuit port, False is just deactivated
+            Whether to replace the deactivated RLC component to a circuit port. The default
+            is ``False``.
```
Comment (PipKat) [inline pyaedt/edb_core/components.py:691]: The first sentence either needs to be one of the following:
"Whether to convert the deactivated RLC component to a circuit port."
"Whether to replace the deactivated RLC component with a circuit port."

Comment (svandenb-dev) [reply pyaedt/edb_core/components.py:691]: Thanks @PipKat :)

**Pull request merged**